utilities/mobile_server_pool.py

```diff
@@ -0,0 +1,227 @@
+import sys
+from requests import Session
+from optparse import OptionParser
+from couchbase.bucket import Bucket
+from couchbase.n1ql import N1QLQuery
+from keywords.utils import log_info
+from couchbase.exceptions import KeyExistsError
+
+SERVER_IP = "172.23.105.177"
+USERNAME = 'Administrator'
+PASSWORD = 'esabhcuoc'
+BUCKET_NAME = "QE-mobile-pool"
+sdk_client = Bucket("couchbase://{}/{}".format(SERVER_IP, BUCKET_NAME))
+
+
+def get_nodes_from_pool_server(num_of_nodes, nodes_os_type, node_os_version, job_name_requesting_node):
+    """
+    Reserve no. of nodes from QE-mobile-pool
+    :param num_of_nodes: No. of nodes one needs
+    :param nodes_os_type: Type of OS for reserve node
+    :param node_os_version: Version of OS for reserve node
+    :param job_name_requesting_node: Name of the job requesting for node
+    :return: list of nodes reserved
+    """
+    # Getting list of available nodes through n1ql query
+    query_str = "select meta().id from `{}` where os='{}' " \
+                "AND os_version='{}' AND state='available'".format(BUCKET_NAME, nodes_os_type, node_os_version)
+    query = N1QLQuery(query_str)
+    pool_list = []
+    for row in sdk_client.n1ql_query(query):
+        doc_id = row["id"]
+        node = reserve_node(doc_id, job_name_requesting_node)
+        if node:
+
+            pool_list.append(str(doc_id))
+        else:
+            continue
+        if len(pool_list) == int(num_of_nodes):
+            return pool_list
+
+    # Not able to allocate all the requested nodes, hence release the node back to the pool
+    release_node(pool_list, job_name_requesting_node)
+    raise Exception("Not enough free node available to satisfy the request")
```

Comment:
you can replace the node with node/nodes as sometime it can be multiple nodes

Reply:
done